library.go

```diff
@@ -1827,6 +1826,49 @@ func string_split(fnname string, recv_ Value, args Tuple, kwargs []Tuple) (Value
 	return NewList(list), nil
 }
 
+func reverse(s string, size int) string {
+	rev := make([]rune, 0, size)
+
+	for _, point := range s {
+		rev = append(rev, point)
+	}
+
+	for i := 0; i < size/2; i++ {
+		rev[i], rev[size-1-i] = rev[size-1-i], rev[i]
+	}
+
+	return string(rev)
+}
+
+func rsplit(s string, max int) []string {
+	size := utf8.RuneCountInString(s)
+	splitCount := 0   // Our current number of splits made.
+	currIndex := size // A backward working index.
+	lastStop := size  // After we make a split, advance this back.
+	parts := make([]string, 0, max)
+
+	reversed := reverse(s, size)
+
+	for _, p := range reversed {
+		if splitCount == max {
+			break
+		}
+		if unicode.IsSpace(p) {
+			parts = append(parts, s[currIndex:lastStop])
+
+			splitCount++
+			lastStop = currIndex - 1
+		}
+		currIndex--
+	}
+
+	res := []string{s[0:lastStop]}
+	for i := len(parts) - 1; i >= 0; i-- {
+		res = append(res, parts[i])
```

Comment:
This allocates a new slice that is essentially the reverse of parts.  It would be more efficient to append the final string to parts, then reverse parts in place (using the logic at lines 1836-38).

Reply:
Good idea, I will change this.